static_files/scripts/fund.sh

```diff
@@ -20,21 +21,25 @@ send() {
 }
 
 for chain_id in "${chain_ids[@]}"; do
-  proposer_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/1")
-  proposer_addr=$(cast wallet address "$proposer_priv")
-  write_keyfile "$proposer_addr" "$proposer_priv" "proposer-$chain_id"
-  batcher_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/2")
-  batcher_addr=$(cast wallet address "$batcher_priv")
-  write_keyfile "$batcher_addr" "$batcher_priv" "batcher-$chain_id"
-  sequencer_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/3")
-  sequencer_addr=$(cast wallet address "$sequencer_priv")
-  write_keyfile "$sequencer_addr" "$sequencer_priv" "sequencer-$chain_id"
-  challenger_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/4")
-  challenger_addr=$(cast wallet address "$challenger_priv")
-  write_keyfile "$challenger_addr" "$challenger_priv" "challenger-$chain_id"
-  send "$proposer_addr"
-  send "$batcher_addr"
-  send "$challenger_addr"
+  for index in "${!roles[@]}"; do
+    role="${roles[$index]}"
+    role_idx=$((index+1))
+
+    # Skip wallet addrs for anything other Proposer/Batcher/Sequencer/Challenger if not on local L1
+    if [[ "${L1_NETWORK}" != "local" && $role_idx -gt 4 ]]; then
+      continue
+    fi
+
+    private_key=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/$role_idx")
+    address=$(cast wallet address "${private_key}")
+    write_keyfile "${address}" "${private_key}" "${role}-$chain_id"
+    send "${address}"
+
+    echo "${role} on chain $chain_id, private key:"${private_key}", address:"${address}""
+  done
 
   cat "/network-data/genesis-$chain_id.json" | jq --from-file /fund-script/gen2spec.jq > "/network-data/chainspec-$chain_id.json"
-done
\ No newline at end of file
+done
+
+echo "L1 faucet private key:"0x${PRIVATE_KEY}", address:"${addr}""
+echo "L2 faucet private key:"0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80", address:"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266""
```

Comment:
For running `tx-fuzzer` for which I had to provide L2 faucet's private key, I looked around and found the faucet private key was hardcoded in a couple of places. Ex: https://github.com/ethereum-optimism/optimism/blob/develop/op-deployer/pkg/deployer/bootstrap/superchain_test.go#L65 . If there's a more elegant way to generate this instead of hardcoding, happy to incorporate.

and in compilations